pkg/network/driver/nmstate/types.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package nmstate
```

Comment:
Can you you please move at least thoe types declaration to a separate commit and explain in the commit message that this structure was taken from [nmstate](https://nmstate.io/devel/yaml_api.html) and whether is was taken as is or changes were done.

Also, please add to the code a comment will a link to the nmstate API.